src/main.js

```diff
@@ -52,6 +52,33 @@ export async function main(platform) {
         });
         await vm.load();
         platform.createAndMountRootView(vm);
+
+        // needed for mobile Safari which does *not* resize the viewport when the keyboard
+        // shows or hides (see https://bugs.webkit.org/show_bug.cgi?id=141832)
+        if (window.visualViewport) {
+            window.visualViewport.addEventListener('resize', () => {
+                const sessionView = document.getElementsByClassName('SessionView')[0];
+                if (!sessionView) {
+                    return;
+                }
+
+                const timeline = document.getElementsByClassName('Timeline')[0];
+                let scrollTopBefore, heightBefore, heightAfter;
+
+                if (timeline) {
+                    scrollTopBefore = timeline.scrollTop;
+                    heightBefore = timeline.getBoundingClientRect().height;
+                }
+
+                sessionView.style.height = window.visualViewport.height.toString() + 'px';
```

Comment:
I'd like to make this workaround obvious in `layout.css` where the [height is set up](https://github.com/vector-im/hydrogen-web/blob/master/src/platform/web/ui/css/layout.css#L46), rather than being an invisible monkey patch in the background. One possibility is to use a css custom property (`--ios-viewport-height`) (support for custom properties overlaps with `visualViewport` support on iOS) to pass the viewport height from JS to CSS, to keep the specifics of which element is resized in CSS. In CSS, you can only apply this custom variable to the height if the root element has the `ios` class set, like we already do for `.legacy` in `spinner.css`. E.g. something like:

```css
.hydrogen.ios .SessionView {
  height: var(--ios-viewport-height, 100%);
}

Reply:
That's pretty ingenious, I hadn't thought of this. Much cleaner than changing the style itself in JS.